### PR TITLE
Perceptual Diff Tests - Stage 2 - Travis CI

### PR DESCRIPTION
**Pull request by DiljotSG**
This PR adds the ability for Travis CI to run the visual difference tests on each pull request. The current state will be compared to the goldens stored in the S3 bucket. Goldens can be regenerated by triggering a custom travis build (See README.md).

### CODE REVIEW

File: test/d2l-quick-eval/perceptual/d2l-quick-eval-search-results-summary-container.visual-diff.js [hunk 1]
```diff
@@ -0,0 +1,35 @@
+const puppeteer = require('puppeteer');
+const VisualDiff = require('@brightspace-ui/visual-diff');
+
+describe('d2l-quick-eval-search-results-summary-container', function() {
+
+	const visualDiff = new VisualDiff('search-results-summary-container', __dirname);
```
Comment (dbatiste) [inline test/d2l-quick-eval/perceptual/d2l-quick-eval-search-results-summary-container.visual-diff.js]: We need to be careful of naming here.  Stay tuned.

File: .travis.yml [hunk 2]
```diff
@@ -1,34 +1,40 @@
...
-  # GITHUB_RELEASE_TOKEN (859f......865e)
-  - secure: QP5YZ7oT1NAyyLDhNG/kHii+SasV5zR/9XO5ePl2ThAevdpEgrLLk1A76NV0XZMMSRIdvgzMZIvPqHWfA81I2O7QCzXOJKQ+xj8hanGqUUNlGxgDMo8FGmqf7OWyxw+XHoM5E0P20yntmDQnCsYnuPvTwYyZ/6ujQgTbPli2ewvoZL+Doh+RnOHmc7R8HbdpU2Cv42PIZr/H/nW6tpuliP1fNrMvKwzjbRu+zh/ag62CkPOk0jbu6klXTwdwytb0jseY7Y/ybJ/0kOZGmgCguH3Lb2kj4Uncp78aQiAZFyCIurb6iXVZ5K7YtABbpWj2sRtMkJwQedTSCUY7PI8FTuvNowZ9URmJ5d4FAAbo03D6jgur4SHxakyV+iQj5z0vfmw5bJT/ykJzXFgyRTkjy/zznZE3arDeyRiLPKGmlKSYDS51eXUhJ2yqgFtsw3ArCqIoIm3xxe9Qy2DNsJllf/SGIhcGhPXIlG4Mj/59OiqNVJuXORJkwuj4SNK4ujkTGkHWYgwGEglhF9nCSPBOrITaFtyfECNvHc9HM/SeYyaDjMqrvXnGusoU6BKhxqiwhb2LhqrK9SRq6ibeP9ykjvi3PkrmbyVHKTAzXHxKJ5uS1orwQ7oOnC4p751pjZB570wIg4ENUsPML/EwbtfsqWjCYnzGZqOgP8mYDXDprr8=
\ No newline at end of file
+language: node_js
+node_js:
+- '11'
```
Comment (AlexBedley) [inline .travis.yml]: Hmm why is this whole file a diff? did you run it through a linter or something? I'm worried the next person that edits this file will cause more churn. Anyways something to watch out for

File: README.md [hunk 3]
```diff
@@ -51,13 +51,31 @@ Quick Eval should be pulled directly from `my-unassessed-activities`:
...
 1. Checkout the master branch: `git checkout master`.
 2. Generate the golden (baseline) images: `npm run test:diff:golden`.
 3. Checkout your branch to test: `git checkout <desired branch>`.
 4. Running the visual difference tests: `npm run test:diff`.
 
+
+#### Procedure for testing on the Travis CI pipeline:
+
+When a pull-request is made, you should consider if your changes will alter the UI and break the visual difference tests.
+If the visuals are being changed/modified, it is necessary for the stored goldens in Amazon S3 to be updated.
```
Comment (AlexBedley) [inline README.md]: How can we view the diff files to see how my PR differs from the current goldens?

File: package.json [hunk 4]
```diff
@@ -40,7 +40,8 @@
     "polymer-cli": "^1.9.2",
     "url-polyfill": "^1.1.3",
     "wct-browser-legacy": "^1.0.1",
-    "whatwg-fetch": "^3.0.0"
+    "whatwg-fetch": "^3.0.0",
+    "puppeteer": "^1"
```
Comment (AlexBedley) [inline package.json]: why is this import necessary?

Comment (dbatiste) [reply package.json]: Not in this particular PR, but `puppeteer` will be imported by the actual tests, since it's the driver.

File: .travis.yml (hunk 2 above)
Comment (DiljotSG) [reply .travis.yml]: Not too sure why that happened.

File: README.md (hunk 3 above)
Comment (DiljotSG) [reply README.md]: The travis CI log will provide a URL that contains the current state compared to the goldens.

**Comment by DiljotSG** [comment]
@AlexBedley Can you re-review this? Stage 3 is ready and I think we should merge that but of course, that depends on the changes in this branch.
